### ISSUE TITLE
Gui restart option in esmda not working properly in the new storage

### ISSUE THREAD

**Issue by hnformentin** [equinor/ert#5737]
**Describe the bug**
User reported issue: https://equinor.slack.com/archives/CDPJULWR4/p1690287924753439
The [manual describes the restart option for esmda](https://fmu-docs.equinor.com/docs/ert/getting_started/gui/gui.html#restarting-es-mda). Nevertheless, the last field "Start iteration" is not available in the gui. I believe it was removed in [this PR](https://github.com/equinor/ert/commit/9167ba8d13fded22cb8ee198b777577438170f92#diff-b3c4f3140b05e74cb1d0548574b4f5e18d52d9254d810350780c834afa3fb4ecL93) for some reason.

**To reproduce**
Steps to reproduce the behaviour:
1.  Try to restart a case as described in the manual. Poly case would make it.

**Expected behaviour**
Expect to have the field available or the manual instructions updated with how to restart.

**Screenshots**
[In the slack thread](https://equinor.slack.com/archives/CDPJULWR4/p1690287924753439).

**Comment by xjules** [comment]
This was the initial [PR](https://github.com/equinor/ert/pull/5052) @hnformentin, hope it might help. The one you are refereeing to came later.

**Comment by hnformentin** [comment]
We went deeper into testing the current behaviour and found a bug in storage implementation.
After running esmda in poly.ert, we tried to restart from iteration 2.
The resulting storage files had all `iteration: 0`.  It generated `default_2_1` (with `default_2` as prior), but also `default_2_2` and `default_2_3`, while we expected only having `default_2_3`.

cat storage/ensembles/*/index.json
{"id": "25cc44ee-f125-471e-9aff-b50fb2d00f5c", "experiment_id": "c12ec6da-cf1d-4ee4-8205-b79fa2e85a86", "ensemble_size": 100, "iteration": 0, "name": "default_2_3", "prior_ensemble_id": "6d667d18-0f4a-4f78-9437-0c2f7e33d472", "started_at": "2023-07-28T09:06:18.280590"}
{"id": "63018d6f-71b4-442e-a4d7-e939948a58b7", "experiment_id": "5348a171-6906-4abb-88df-08d8481594c9", "ensemble_size": 100, "iteration": 0, "name": "default", "prior_ensemble_id": null, "started_at": "2023-07-28T08:33:49.104512"}
{"id": "6d667d18-0f4a-4f78-9437-0c2f7e33d472", "experiment_id": "c12ec6da-cf1d-4ee4-8205-b79fa2e85a86", "ensemble_size": 100, "iteration": 0, "name": "default_2_2", "prior_ensemble_id": "965430a2-fdfa-4908-9de9-3424684faa30", "started_at": "2023-07-28T09:05:57.508289"}
{"id": "81bcf783-f985-4825-9b8e-02071b62a3a0", "experiment_id": "a6687d57-bd48-4cb8-8519-ba8e351080db", "ensemble_size": 100, "iteration": 0, "name": "default_2", "prior_ensemble_id": "fa0d23e8-d1b0-4dea-a4a9-9167a5cf2d4d", "started_at": "2023-07-28T08:34:50.846798"}
{"id": "965430a2-fdfa-4908-9de9-3424684faa30", "experiment_id": "c12ec6da-cf1d-4ee4-8205-b79fa2e85a86", "ensemble_size": 100, "iteration": 0, "name": "default_2_1", "prior_ensemble_id": "81bcf783-f985-4825-9b8e-02071b62a3a0", "started_at": "2023-07-28T09:05:31.917355"}
{"id": "e902f708-5f6f-4dd7-aa82-737ccb970581", "experiment_id": "a6687d57-bd48-4cb8-8519-ba8e351080db", "ensemble_size": 100, "iteration": 0, "name": "default_3", "prior_ensemble_id": "81bcf783-f985-4825-9b8e-02071b62a3a0", "started_at": "2023-07-28T08:35:09.608193"}
{"id": "f11ce554-6d7d-44c5-9786-6637703c8d1f", "experiment_id": "a6687d57-bd48-4cb8-8519-ba8e351080db", "ensemble_size": 100, "iteration": 0, "name": "default_0", "prior_ensemble_id": null, "started_at": "2023-07-28T08:34:09.651522"}
{"id": "fa0d23e8-d1b0-4dea-a4a9-9167a5cf2d4d", "experiment_id": "a6687d57-bd48-4cb8-8519-ba8e351080db", "ensemble_size": 100, "iteration": 0, "name": "default_1", "prior_ensemble_id": "f11ce554-6d7d-44c5-9786-6637703c8d1f", "started_at": "2023-07-28T08:34:29.551585"}
